kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
@dessant when touch_multiselect=False and self.multiselect=True , then to deselect any single selected node one has to (press + ctrl_down) to deselect it (that was the feature earlier) . But if u want that on clicking on an already selected node should clear all selections including the one which is clicked upon then simply removing the line 315 works fine.... I think i was wrong earlier then.